### ISSUE TITLE
False positive `no-undef` for template-imports and contextual components

### ISSUE THREAD

**Issue by simonihmig** [ember-cli/eslint-plugin-ember#1747]
This is failing linting with `form` and `field` "locals" (they are not local to the outer template scope, so _not_ supposed to be provided by JS scope, which ESLint thinks here) falsely violating `no-undef`:

```gjs
<template>
  <HeadlessForm @data={{data}} as |form|>
    <form.field @name="firstName" as |field|>
      <field.label>First Name</field.label>
      <field.input data-test-first-name />
    </form.field>
    <form.field @name="lastName" as |field|>
      <field.label>Last Name</field.label>
      <field.input data-test-last-name />
    </form.field>
  </HeadlessForm>
</template>
```

I think I have identified the root cause in ` @glimmer/syntax`, keeping this for tracking. /cc @NullVoxPopuli

**Comment by simonihmig** [comment]
My attempt to fix this upstream: https://github.com/glimmerjs/glimmer-vm/pull/1406

**Comment by bmish** [comment]
Note that this change was reverted:

* https://github.com/ember-cli/eslint-plugin-ember/pull/1920
* https://github.com/ember-cli/eslint-plugin-ember/pull/1941
